### PR TITLE
curvefs/client: add metric for client

### PR DESCRIPTION
**Pull request by ilixiaocui**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/fuse_client.cpp [hunk 1]
```diff
@@ -166,6 +183,8 @@ CURVEFS_ERROR FuseClient::FuseOpInit(
     LOG(INFO) << "Mount " << fsName << " on " << mountPointWithHost
               << " success!";
     init_ = true;
+
+    fsMetric_.fsName = fsName;
```
Comment (xu-chaojie) [inline curvefs/src/client/fuse_client.cpp:187]: move it above ”init_ = true;“

File: curvefs/src/client/metric/client_metric.h [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+struct FSMetric {
+    const std::string prefix = "curvefs_client";
+
+    std::string fsName;
+
+    InterfaceMetric userWrite;
+    InterfaceMetric userRead;
+
+    explicit FSMetric(const std::string &name = "")
```
Comment (xu-chaojie) [inline curvefs/src/client/metric/client_metric.h:106]: FSMetric is always construct before fsName is inited.

File: curvefs/src/client/rpcclient/mds_client.cpp [hunk 3]
```diff
@@ -141,12 +146,14 @@ FSStatusCode MdsClientImpl::MountFs(const std::string& fsName,
...
-FSStatusCode MdsClientImpl::UmountFs(const std::string& fsName,
-                                     const std::string& mountPt) {
+FSStatusCode MdsClientImpl::UmountFs(const std::string &fsName,
+                                     const std::string &mountPt) {
     auto task = RPCTask {
+        mdsClientMetric_.umountFs.qps.count << 1;
         UmountFsResponse response;
         mdsbasecli_->UmountFs(fsName, mountPt, &response, cntl, channel);
         if (cntl->Failed()) {
+            mdsClientMetric_.umountFs.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/mds_client.cpp:156]: Should we cout eps when ret != FSStatusCode::OK like MountFs？

File: curvefs/src/client/rpcclient/mds_client.cpp [hunk 4]
```diff
@@ -162,12 +169,15 @@ FSStatusCode MdsClientImpl::UmountFs(const std::string& fsName,
...
-                                      FsInfo* fsInfo) {
+FSStatusCode MdsClientImpl::GetFsInfo(const std::string &fsName,
+                                      FsInfo *fsInfo) {
     auto task = RPCTask {
+        mdsClientMetric_.getFsInfo.qps.count << 1;
         GetFsInfoResponse response;
         mdsbasecli_->GetFsInfo(fsName, &response, cntl, channel);
+
         if (cntl->Failed()) {
+            mdsClientMetric_.getFsInfo.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/mds_client.cpp:180]: Should we cout eps when ret != FSStatusCode::OK like MountFs？

File: curvefs/src/client/rpcclient/mds_client.cpp [hunk 5]
```diff
@@ -203,6 +214,7 @@ FSStatusCode MdsClientImpl::GetFsInfo(uint32_t fsId, FsInfo* fsInfo) {
 
         FSStatusCode ret = response.statuscode();
         if (ret != FSStatusCode::OK) {
+            mdsClientMetric_.getFsInfo.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/mds_client.cpp:217]: same as above

File: curvefs/src/client/rpcclient/mds_client.cpp [hunk 6]
```diff
@@ -214,12 +226,15 @@ FSStatusCode MdsClientImpl::GetFsInfo(uint32_t fsId, FsInfo* fsInfo) {
...
-    const std::vector<PartitionTxId>& txIds) {
+TopoStatusCode
+MdsClientImpl::CommitTx(const std::vector<PartitionTxId> &txIds) {
     auto task = RPCTask {
+        mdsClientMetric_.commitTx.qps.count << 1;
         CommitTxResponse response;
         mdsbasecli_->CommitTx(txIds, &response, cntl, channel);
+
         if (cntl->Failed()) {
+            mdsClientMetric_.commitTx.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/mds_client.cpp:237]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 7]
```diff
@@ -82,6 +83,7 @@ MetaStatusCode MetaServerClientImpl::GetDentry(uint32_t fsId, uint64_t inodeid,
         stub.GetDentry(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.getDentry.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:86]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 8]
```diff
@@ -140,6 +144,7 @@ MetaStatusCode MetaServerClientImpl::ListDentry(uint32_t fsId, uint64_t inodeid,
         stub.ListDentry(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.listDentry.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:147]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 9]
```diff
@@ -205,6 +211,7 @@ MetaStatusCode MetaServerClientImpl::CreateDentry(const Dentry &dentry) {
                 << oss.str();
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.createDentry.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:214]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 10]
```diff
@@ -261,6 +269,7 @@ MetaStatusCode MetaServerClientImpl::DeleteDentry(uint32_t fsId,
         stub.DeleteDentry(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.deleteDentry.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:272]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 11]
```diff
@@ -311,6 +321,7 @@ MetaServerClientImpl::PrepareRenameTx(const std::vector<Dentry> &dentrys) {
         stub.PrepareRenameTx(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.prepareRenameTx.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:324]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 12]
```diff
@@ -360,6 +372,7 @@ MetaStatusCode MetaServerClientImpl::GetInode(uint32_t fsId, uint64_t inodeid,
         stub.GetInode(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.getInode.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:375]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 13]
```diff
@@ -425,6 +439,7 @@ MetaStatusCode MetaServerClientImpl::UpdateInode(const Inode &inode) {
         stub.UpdateInode(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.updateInode.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:442]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 14]
```diff
@@ -477,6 +493,7 @@ MetaStatusCode MetaServerClientImpl::CreateInode(const InodeParam &param,
         stub.CreateInode(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.createInode.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:496]: same as above

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 15]
```diff
@@ -531,6 +549,7 @@ MetaStatusCode MetaServerClientImpl::DeleteInode(uint32_t fsId,
         stub.DeleteInode(cntl, &request, &response, nullptr);
 
         if (cntl->Failed()) {
+            metaserverClientMetric_.deleteInode.eps.count << 1;
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:552]: same as above

File: curvefs/src/client/rpcclient/mds_client.cpp (hunk 3 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/mds_client.cpp:156]: this is rpc error, FSStatsCode not ok is logic.

**Review by baijiaruo — APPROVED** [review]

File: curvefs/src/client/fuse_client.cpp (hunk 1 above)
Comment (ilixiaocui) [reply curvefs/src/client/fuse_client.cpp:187]: done

File: curvefs/src/client/metric/client_metric.h (hunk 2 above)
Comment (ilixiaocui) [reply curvefs/src/client/metric/client_metric.h:106]: yes, we will init fsName during mount.

File: curvefs/src/client/rpcclient/mds_client.cpp (hunk 4 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/mds_client.cpp:180]: ditto

File: curvefs/src/client/rpcclient/mds_client.cpp (hunk 5 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/mds_client.cpp:217]: ditto

File: curvefs/src/client/rpcclient/mds_client.cpp (hunk 6 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/mds_client.cpp:237]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 7 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:86]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 8 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:147]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 9 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:214]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 10 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:272]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 11 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:324]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 12 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:375]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 13 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:442]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 14 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:496]: ditto

File: curvefs/src/client/rpcclient/metaserver_client.cpp (hunk 15 above)
Comment (ilixiaocui) [reply curvefs/src/client/rpcclient/metaserver_client.cpp:552]: ditto

**Comment by ilixiaocui** [comment]
recheck

**Comment by ilixiaocui** [comment]
recheck

**Comment by ilixiaocui** [comment]
recheck

**Comment by ilixiaocui** [comment]
recheck

**Comment by ilixiaocui** [comment]
recheck

**Review by xu-chaojie — APPROVED** [review]

**Pull request merged**